src/plugins/telemetry/public/services/telemetry_sender.ts

```diff
@@ -80,17 +89,18 @@ export class TelemetrySender {
             })
         )
       );
-      this.lastReported = `${Date.now()}`;
-      this.saveToBrowser();
     } catch (err) {
-      // ignore err
-    } finally {
-      this.isSending = false;
+      // ignore err and try again but after a longer wait period.
+      this.retryCount = this.retryCount + 1;
+      if (this.retryCount < 20) {
+        // exponentially backoff the time between subsequent retries to up to 19 attempts, after which we give up until the next report is due
+        window.setTimeout(this.sendUsageData, this.delayRetrySend());
```

Comment:
NIT/optional: `delayRetrySend` can be extracted as a static function `getRetryDelay(retryCount: number)`

Reply:
Excellent suggestion, thank you! It also made the testing much easier.